src/gc/gc.cpp

```diff
@@ -13198,17 +13329,26 @@ BOOL gc_heap::allocate_large (int gen_number,
     if (loh_alloc_state == a_state_cant_allocate)
     {
         assert (oom_r != oom_no_failure);
-        handle_oom (heap_number, 
-                    oom_r, 
-                    size,
-                    0,
-                    0);
-
+        if (should_retry_other_heap (size))
```

Comment:
should be
        if ((oom_r != oom_cant_commit) && should_retry_other_heap (size))